### PR TITLE
PR template when adding a new library

### PR DESCRIPTION
**Pull request by Wauplin**
This PR adds a template with a checklist to follow when adding a new library.

Based on https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository.

Once merged, we should be able to generate a link to create such a PR (the template will not be used by default for all PRs :smile:). This link would be added to https://huggingface.co/docs/hub/models-download-stats#can-i-add-my-query-files-for-my-library.

Hopefully, this should reduce the back-and-fourth when someone adds a new library. I suggest we start with that as an experiment and if it doesn't work, we can move it somewhere else as documentation.

### CODE REVIEW

File: .github/pull_request_template/new_library.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+- [ ] `repoName` and `prettyLabel` are set with user-friendly casing (example: `DeepForest`).
+- [ ] `repoUrl` is set with a link to the library source code (usually a GitHub repository).
+- [ ] (optional) `docsUrl` is set with a link to the docs of the library. If the documentation is in the GitHub repo referenced above, no need to set it twice.
+- [ ] `filter` is set to `false`.
+- [ ] `countDownload` follows the correct convention and do not duplicate the counting. For instance, if loading a model requires 3 files, the download count rule must count downloads only on 1 of the 3 files. Otherwise, the download count will be overestimated. If the library uses one of the default config files (`config.json`, `config.yaml`, `hyperparams.yaml`, and `meta.yaml`, see [here](https://huggingface.co/docs/hub/models-download-stats#which-are-the-query-files-for-different-libraries)), there is no need to manually define a download count rule.
+- [ ] (optional) `snippets` is correctly defined. See check-list below.
```
Comment (Vaibhavs10) [inline .github/pull_request_template/new_library.md:22]: Can we add "make sure that the code passes the Linter test as well" haha!

**Review by Vaibhavs10 — APPROVED** [review]
Thanks a lot for opening this! Indeed is needed.

I just wonder if we need the `snippets` bit as not a lot of libraries have it and seeing a wall of text as the first thing might deter them from making a contribution WDYT?

**Review by julien-c — COMMENTED** [review]
looks nice!

File: .github/pull_request_template/new_library.md (hunk 1 above)
Comment (Wauplin) [reply .github/pull_request_template/new_library.md:22]: good idea! Added in https://github.com/huggingface/huggingface.js/pull/913/commits/b54ffcce546f703f087e071300212e00a579559c

**Comment by Wauplin** [comment]
I've addressed feedback in https://github.com/huggingface/huggingface.js/pull/913/commits/b54ffcce546f703f087e071300212e00a579559c. I have added `"(optional)"` in the snippets title to make it clear it's optional. Let's merge it as it is and reassess after if needed.

**Comment by Wauplin** [comment]
Well, I've been enthusiastic a bit too quickly... This feature requires the user to manually edit their URL to make it work >< . I'll try to hack something based on https://stackoverflow.com/questions/73771068/multiple-templates-for-pull-requests-on-github

**Pull request merged**